--- abrt-1.0.9/lib/Utils/DebugDump.cpp~	2010-04-02 12:53:59.000000000 +0300
+++ abrt-1.0.9/lib/Utils/DebugDump.cpp	2010-04-08 23:04:27.265361307 +0300
@@ -331,6 +331,7 @@
         SaveText(FILENAME_ARCHITECTURE, buf.machine);
         std::string release;
         LoadTextFile("/etc/redhat-release", release);
+        LoadTextFile("/etc/pld-release", release);
         const char *release_ptr = release.c_str();
         unsigned len_1st_str = strchrnul(release_ptr, '\n') - release_ptr;
         release.erase(len_1st_str); /* usually simply removes trailing '\n' */
