diff -urNp -x '*.orig' abrt-2.14.4.org/src/configuration-gui/Makefile.am abrt-2.14.4/src/configuration-gui/Makefile.am
--- abrt-2.14.4.org/src/configuration-gui/Makefile.am	2020-08-17 16:37:07.000000000 +0200
+++ abrt-2.14.4/src/configuration-gui/Makefile.am	2021-03-02 22:08:20.452408463 +0100
@@ -36,6 +36,7 @@ libabrt_gui_la_LIBADD = \
 	../lib/libabrt.la \
 	$(GLIB_LIBS) \
 	$(GTK_LIBS) \
+	../lib/libabrt.la \
 	$(LIBREPORT_LIBS)
 
 bin_PROGRAMS = system-config-abrt
