Allow rpm 4.17.x (rpmhex was introduced in 4.18, pgpHexStr removed in 4.19?)
--- abrt-2.17.5/src/daemon/rpm.c.orig	2024-02-18 14:55:16.000000000 +0100
+++ abrt-2.17.5/src/daemon/rpm.c	2024-03-10 17:34:33.751952467 +0100
@@ -113,7 +113,7 @@ void rpm_load_gpgkey(const char* filenam
     pubkey = rpmPubkeyNew(pkt, pklen);
     if (pubkey != NULL)
     {
-        fingerprint = rpmhex(pubkey->keyid, sizeof(pubkey->keyid));
+        fingerprint = pgpHexStr(pubkey->keyid, sizeof(pubkey->keyid));
         if (fingerprint != NULL)
             list_fingerprints = g_list_append(list_fingerprints, fingerprint);
 
@@ -123,7 +123,7 @@ void rpm_load_gpgkey(const char* filenam
             rpmPubkey subkey = subkeys[i];
             if (subkey != NULL)
             {
-                fingerprint = rpmhex(subkey->keyid, sizeof(subkey->keyid));
+                fingerprint = pgpHexStr(subkey->keyid, sizeof(subkey->keyid));
                 if (fingerprint != NULL)
                     list_fingerprints = g_list_append(list_fingerprints, fingerprint);
             }
