--- abrt-2.1.7/src/python-problem/problem/Makefile.am.orig	2013-09-10 14:11:13.000000000 +0200
+++ abrt-2.1.7/src/python-problem/problem/Makefile.am	2013-09-27 19:40:42.648094557 +0200
@@ -1,6 +1,6 @@
 problem_PYTHON = __init__.py exception.py proxies.py tools.py watch.py config.py
 
-problemdir = $(pythondir)/problem
+problemdir = $(pyexecdir)/problem
 
 pyabrtdir = $(problemdir)
 pyabrt_LTLIBRARIES = _pyabrt.la
