--- abrt-2.0.15/configure.ac.orig	2012-09-24 09:43:03.000000000 +0200
+++ abrt-2.0.15/configure.ac	2012-10-14 23:06:30.028035439 +0200
@@ -3,7 +3,7 @@
 	[crash-catcher@fedorahosted.org])
 
 AC_CONFIG_MACRO_DIR([m4])
-AM_INIT_AUTOMAKE([-Wall -Werror foreign])
+AM_INIT_AUTOMAKE([-Wall foreign])
 # Support silent build rules. Disable by either passing --disable-silent-rules
 # to ./configure or passing V=1 to make
 m4_ifdef([AM_SILENT_RULES], [AM_SILENT_RULES])
@@ -11,10 +11,12 @@
 AM_SILENT_RULES([yes])
 
 
+AM_PROG_AR
 AC_DISABLE_STATIC
 AC_PROG_LIBTOOL
 AC_PROG_CC
 AC_PROG_LN_S
+AM_PROG_CC_C_O
 
 AC_SYS_LARGEFILE
 CFLAGS="$CFLAGS -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE \
