--- abrt-2.1.3/src/plugins/abrt-retrace-client.c.orig	2013-04-02 17:15:14.012788254 +0200
+++ abrt-2.1.3/src/plugins/abrt-retrace-client.c	2013-04-02 17:42:57.132753624 +0200
@@ -741,7 +741,7 @@
     if (response_code == 500 || response_code == 507)
     {
         alert_server_error(cfg.url);
-        error_msg_and_die(http_body);
+        error_msg_and_die("%s", http_body);
     }
     else if (response_code == 403)
     {
